lib/Parse/ParseDecl.cpp

```diff
@@ -4148,6 +4151,15 @@ static unsigned skipUntilMatchingRBrace(Parser &P,
     HasNestedTypeDeclarations |= P.Tok.isAny(tok::kw_class, tok::kw_struct,
                                              tok::kw_enum);
 
+
+    HasPotentialRegexLiteral |= (BareSlashRegexLiterals &&
+                                 P.Tok.isAnyOperator() &&
+                                 P.Tok.getText().contains('/'));
```

Comment:
One thing we could potentially do here is check for a space after the `/`, which should allow us to exclude many uses of e.g infix `/` (as a regex literal cannot start with space). If we really wanted to, we could potentially split off the heuristics from `Lexer::tryLexRegexLiteral` and use those to narrow down the cases we consider